File: ICARUS/Models/Icarus/Elements/NAVSEPARATOR.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ICARUS.Models.Icarus.Elements {

    /// <summary>
    /// Represents a horizontal line that seperates groupings inside of a dropdown menu
    /// </summary>
    public class NAVSEPARATOR : NAVITEM {

        /// <summary>
        /// Constructs a list item (LI) to be added to a list (UL > LI)
        /// </summary>
        /// <param name="attributes"></param>
        /// <param name="label">The text to be displayed within the LI tags</param>
        public NAVSEPARATOR() : base() {
            this.attributes.Add("role", "separator");
            this.attributes.Add("class", "divider");
        }
    }
}

File: ICARUS/Scripts/icarus/IcarusNavBar/NAVSEPARATOR.js
/**
    A horizontal line that separates content inside a navbar dropdown menu
*/
class NAVSEPARATOR extends LI {
    /**
        A separator for menus
        @param {EL} node The element that will contain this object
     */
    constructor(node) {
        super(node, new MODEL(new ATTRIBUTES({
            'role': 'separator',
            'class': 'divider nav-item'
        })));
    }    
}